src/libraries/System.Private.CoreLib/src/System/Environment.Browser.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System
+{
+    public static partial class Environment
+    {
+        // Emscripten VFS mounts at / and is the only drive
+        public static string[] GetLogicalDrives() => new string[] { "/" };
+
+        // In the mono runtime, this maps to gethostname, which returns 'emscripten'.
+        // Returning the value here allows us to exclude more of the runtime.
+        public static string MachineName => "emscripten";
```

Comment:
I'm not sure we want to expose this value

@lewing

Reply:
I'd rather have it be "localhost" or better yet map, if we can, to location.hostname.  But this value is compatible to what's returned in 3.2

Reply:
I'm not concern about this one, we are switching from Mono BCL to .NET Core anyway

Reply:
The Unix implementation already does:
```
Interop.Sys.GetHostName();
```
and SystemNative_GetHostName just calls:
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/Native/Unix/System.Native/pal_networking.c#L573
Why doesn't that work for wasm?

Reply:
It does work, it just returns "emscripten".  We may be able to link something out by hard coding it into Environment.

Reply:
@stephentoub we cannot remove unused native code, calling pinvoke means every app will have the code included, in this case, to return a constant string which is not worth it.

Reply:
Interop.Sys.GetHostName is used by Environment, by Dns, by PipeStream, by NetworkInformation, and by Process.  We're hardcoding all such call sites?  Or none of those are available on browser and this is the only one left?

Reply:
I'll leave that to @steveisok to confirm but I think we turned all these assemblies into PNSE on browser

Reply:
Correct, all of them except Dns are already PNSE and I'm working on doing that for Dns as well.

Reply:
Ok, if we're going to hardcode it, I agree we should use something other than "emscripten".  Why match what GetHostName returns if we're trying not to use GetHostName, and if we're trying to hide the fact that it's implemented as unix under the covers with emscripten.  "localhost" sounds reasonable to me, or ".", or even empty string.